README.md

```diff
@@ -454,13 +454,25 @@ a multi-architecture
 (aka "manifest list"), with support for each OS and architecture pair supported
 by the base image.
 
-If `ko` is invoked with `--platform=<some-OS>/<some-platform>` (e.g.,
+If `ko` is invoked with `--platform=<some-OS>/<some-arch>` (e.g.,
 `--platform=linux/amd64` or `--platform=linux/arm64`), then it will attempt to
 build an image for that OS and architecture only, assuming the base image
 supports it.
 
-When `--platform` is not provided, `ko` builds an image with the OS and
-architecture based on the build environment's `GOOS` and `GOARCH`.
+If the base image is a manifest list with more platforms than you want to build,
+invoking `ko` with comma-separated list of platforms (e.g.
+`--platform=linux/amd64,linux/arm/v6`) will produce a manifest list
+containing only the provided platforms. Note that if the base image does not
+contain platforms that are provided by this flag, `ko` will be unable to build
+a corresponding image, and this is not an error. The resulting artifact will be
+a multi-platform image containing the intersection of platforms from the base
+image and the `--platform` flag. This is especially relevant for projects that
+use multiple base images, as you must ensure that every base image contains all
+the platforms that you'd like to build.
+
+When `--platform` is not provided, if both `GOOS` and `GOARCH` environment
+variables are set, `ko` will build an image for `${GOOS}/${GOARCH}`, otherwise
+`ko` will build a `linux/amd64` image.
```

Comment:
should thus mention GOARM?

Reply:
I don't think so? Defaulting doesn't consider GOARM at all.

Reply:
I wonder if it should 🤔

Reply:
Oh you're saying we can use `GOARM` to break platform.Variant ties in base image resolution if all of GOOS/GOARCH/GOARM are set? Not a bad idea.

Reply:
Though... a little ambiguous. Currently local env overrides the platform from the base image (GOOS and GOARCH) [here](https://github.com/google/ko/blob/960e6a3f1c039d3b7635c7834b10dab59d548b2a/pkg/build/gobuild.go#L259-L274).

So we can't use GOARM for both selection and overriding in every context. I'm having a hard time wrapping my head around how the defaulting, selection, and overriding precedence rules should work.

I guess maybe if you say `--platform=linux/arm` and set `GOARM=6`, we would take the first `linux/arm` image and always use `GOARM=6`, but if you don't set `--platform` and _do_ set all of GOOS/GOARCH/GOARM, we will use that whole thing as our default platform? We now need to map from `GOARM` to `platform.Variant` (just prepend a `v`, I guess?).

Reply:
Just a reminder that if these questions seem too complex to answer, we can still just not support comma-delimited archs and still only support `--all`, until such a time as we have a user that wants it and can tell us how they'd like to use it.

Reply:
Alex Ellis was asking, and knative/serving would probably also be fine with restricting multi-arch to amd64/arm64 since Envoy only ships those.

Reply:
Sure. For Tekton at least, I think we're fine producing images that happen to work on some archs because we produce them with `--platform=all`, but we won't commit to supporting them unless explicitly stated otherwise. Knative etc could follow the same path.

In particular this makes it easier to experiment with supporting new archs since it's already built and released using `--platform=all`, and you don't have to add `--platform=foo,bar,new-thing`.

Reply:
Yeah, that's where we are now.  I was just looking to save some trees in places where Envoy didn't work 😉

Reply:
>  these questions seem too complex to answer

I don't know if they're particularly complex to answer. Any behavior we pick is fine, IMO, but I'm trying to determine what the least surprising thing would be.

Reply:
> So we can't use GOARM for both selection and overriding in every context. I'm having a hard time wrapping my head around how the defaulting, selection, and overriding precedence rules should work.

It was true: https://github.com/google/ko/issues/274

Reply:
Done.